examples/fedrecon_no_mpi.py

```diff
@@ -0,0 +1,156 @@
+    - each column of A is associated with pixel. In this example, there are 250*250 pixels.
+    """
+
+    mat_path = os.getcwd() + "/datasets/RawData/fedrecon/%s.mat" % (args.dataset)
+
+    data = scipy.io.loadmat(mat_path)
+    A = torch.from_numpy(data["A"].toarray()).to(torch.float32)
+    b1 = torch.from_numpy(data["b1"]).to(torch.float32)
+    b2 = torch.from_numpy(data["b2"]).to(torch.float32)
+    b3 = torch.from_numpy(data["b3"]).to(torch.float32)
+    b = torch.from_numpy(data["b"]).to(torch.float32)
+
+    ## coefficient assigned for each client
+    c = torch.from_numpy(data["c"][0]).to(torch.float32)
+    for idx in range(len(c)):
+        cfg.fed.args.coeff[idx] = c[idx].item()
+    ## ground truth (to compute the mean squared error for every iterations)
+    w1_truth = data["w1"].transpose().reshape(-1).tolist()
+    w2_truth = data["w2"].transpose().reshape(-1).tolist()
+    w3_truth = data["w3"].transpose().reshape(-1).tolist()
+    w_truth = data["w"].transpose().reshape(-1).tolist()
+
+    for idx in range(len(w1_truth)):
+        cfg.fed.args.w_truth[0].append(w1_truth[idx])
+        cfg.fed.args.w_truth[1].append(w2_truth[idx])
+        cfg.fed.args.w_truth[2].append(w3_truth[idx])
+        cfg.fed.args.w_truth[3].append(w_truth[idx])
+
+    train_datasets = []
+    train_datasets.append(Dataset(A, b1))
+    train_datasets.append(Dataset(A, b2))
+    train_datasets.append(Dataset(A, b3))
+    train_datasets.append(Dataset(A, b))
+
+    return train_datasets
+
+
+## Run
+def main():
+    """Configuration"""
+    cfg = OmegaConf.structured(Config)
+    cfg.train_data_shuffle=False
+    cfg.device = args.device
+    cfg.reproduce = True
+    if cfg.reproduce == True:
+        set_seed(1)
+
+    ## Turn off save models for all client
+    cfg.save_model_state_dict = False
+
+    ## outputs
+    cfg.output_dirname = "./outputs_%s" % (args.dataset)
+    cfg.output_filename = "result" 
+    
+    ## algorithm
+    cfg.fed = eval(args.federation_type + "()")
+
+    ## server
+    cfg.num_epochs = args.num_epochs
+
+    ## clients
+    cfg.num_clients = args.num_clients
+    cfg.fed.args.optim = args.client_optimizer
+    cfg.fed.args.optim_args.lr = args.client_lr
+    cfg.fed.args.num_local_epochs = args.num_local_epochs
+    cfg.batch_training = args.batch_training
+    cfg.train_data_batch_size = args.train_data_batch_size
+
+    ## tensorboard
+    cfg.fed.args.t_filename="%s_%s" % (args.dataset, args.federation_type)    
+
+    """ User-defined data """
+    train_datasets = get_data(cfg)
+
+    """ User-defined model """
+    num_feature = train_datasets[0][0][0].size(dim=0)
+    model = LinearRegression(num_feature, 1)
+    loss_fn = torch.nn.MSELoss(reduction="sum")
+
+    """ Running """
+    rs.run_serial(cfg, model, loss_fn, train_datasets, Dataset(), args.dataset)
```

Comment:
I wonder whether there is any specific reason only using serial run, or just for easy testing

Reply:
Just for easy testing. In this specific application, the number of clients is fixed to 4, so I think a serial run is good enough for now.